nessie/jobs/scheduling.py

```diff
@@ -27,31 +27,45 @@
 """Background job scheduling."""
 
 
+from apscheduler.jobstores.sqlalchemy import SQLAlchemyJobStore
 from apscheduler.schedulers.background import BackgroundScheduler
 
 
+app = None
 sched = None
 
 
+# Postgres advisory locks require numeric ids.
+PG_ADVISORY_LOCK_IDS = {
+    'JOB_SYNC_CANVAS_SNAPSHOTS': 1000,
+    'JOB_RESYNC_CANVAS_SNAPSHOTS': 2000,
+    'JOB_GENERATE_ALL_TABLES': 3000,
+}
+
+
 def get_scheduler():
     return sched
 
 
-def initialize_job_schedules(app):
+def initialize_job_schedules(_app):
     from nessie.jobs.create_canvas_schema import CreateCanvasSchema
     from nessie.jobs.create_sis_schema import CreateSisSchema
     from nessie.jobs.generate_boac_analytics import GenerateBoacAnalytics
     from nessie.jobs.generate_intermediate_tables import GenerateIntermediateTables
     from nessie.jobs.resync_canvas_snapshots import ResyncCanvasSnapshots
     from nessie.jobs.sync_canvas_snapshots import SyncCanvasSnapshots
 
+    global app
+    app = _app
```

Comment:
`app` has changed from a passed-around argument to another dicey global because with the move to SQLAlchemyJobStore, job arguments must be serializable to binary blobs via `pickle`.

Reply:
Thanks for trying out some alternatives. Maybe we'll be able to get serializable arguments some other way later, but this shouldn't block the merge.